src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  die(dead) {
+    if (dead.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && !(animal instanceof Carnivore)) {
```

Comment:
[FUNCTION NAMING] The condition in the `bite` method should only check if the `animal` is an instance of `Herbivore` and not hidden. The current implementation also checks if the `animal` is not an instance of `Carnivore`, which is redundant since `Herbivore` and `Carnivore` are distinct classes and a `Carnivore` cannot be an instance of `Herbivore`.